packages/ruby/lib/readme/metrics.rb

```diff
@@ -39,11 +39,26 @@ def call(env)
       start_time = Time.now
       status, headers, body = @app.call(env)
       end_time = Time.now
-
       response = Rack::Response.new(body, status, headers)
 
-      har = Har::Serializer.new(env, response, start_time, end_time, @filter)
+      begin
+        process_response(
+          response: response,
+          env: env,
+          start_time: start_time,
+          end_time: end_time
+        )
+      rescue
+        [status, headers, body]
+      end
```

Comment:
Would it make sense to move the rescue into `process_response`?

Reply:
I considered that, but it seemed to be easier to test this way because we can make `process_response` raise without having to worry about what part of it raises.

Reply:
Hmmm, I hadn't noticed we were stubbing that. I'd prefer not to stub the system under test and instead stub a collaborator. That results in tests that are less brittle and less likely to be tautologies. Thoughts on having either `Payload#to_json` or `RequestQueue#push` raise in the tests instead?

Reply:
That sounds reasonable. Something about choosing just one of the collaborators feels incomplete. Does it make sense to test a number of them? Or all of them? Or is one sufficient?

Reply:
I would feel confident with just one of the collaborators but if you want to have a test case for each of them that's fine too